src/app/services/school-requests/school-request.service.ts

```diff
@@ -2,29 +2,31 @@ import { apiUrl } from 'src/constants/urls';
 import { ISchoolRequestVM } from '../../components/School Requests/SchoolRequest/ISchoolRequestVM';
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable, from } from 'rxjs';
 import { HttpOptionsService } from '../http-options/http-options.service';
 
 @Injectable({
   providedIn: 'root'
 })
-export class SchoolRequestService {
+export class SchoolRequestsService {
 
-  private url: string = apiUrl+"/SchoolRequestProccesing";
-  constructor(private http:HttpClient,private httpOptions:HttpOptionsService) { }
+  private url: string = apiUrl + "/SchoolRequests";
+  constructor(private http:HttpClient, private httpOptions:HttpOptionsService) { }
+  requests:ISchoolRequestVM[];
 
-  get = (): Observable<ISchoolRequestVM[]> => {
+  get = () => {
     this.httpOptions.loadHeaders();
-    return this.http.get<ISchoolRequestVM[]>(this.url,this.httpOptions.options);
+    return this.http.get(this.url,this.httpOptions.options).subscribe( (res: ISchoolRequestVM[]) => {
+      this.requests=res;
+    } );
   };
 
-  approve = (id:number): Observable<any> => {
+  approve = (id:number) => {
     this.httpOptions.loadHeaders();
-    return this.http.post<any>(this.url,{id:id},this.httpOptions.options);
+    return this.http.post(this.url+"/Approve",{id:id},this.httpOptions.options).subscribe((res) => {this.get()});
   };
```

Comment:
Spaces.

Reply:
each one?

Reply:
You need to add 9 spaces to this piece of code.